File: Modules/Kit/MES/SIE.Web.Kit.MES/CallMaterials/Commands/SetUrgencyCommand.cs
using SIE.Kit.MES.CallMaterials;
using SIE.Web.Command;

namespace SIE.Web.Kit.MES.CallMaterials.Commands
{
    /// <summary>
    /// 设置紧急
    /// </summary>
    [JsCommand("SIE.Web.Kit.MES.CallMaterials.Commands.SetUrgencyCommand")]
    public class SetUrgencyCommand : ViewCommand<ViewArgs>
    {
        /// <summary>
        /// 执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">SIE.MetaModel.View.Block.EntityType</param>
        /// <returns>object</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var bill = args.Data.ToJsonObject<CallMaterialBill>();
            RT.Service.Resolve<CallMaterialController>().SetUrgency(bill.Id);
            return "操作成功";
        }
    }
}

File: Modules/Kit/MES/SIE.Web.Kit.MES/CallMaterials/Commands/SetUrgencyCommand.js
SIE.defineCommand('SIE.Web.Kit.MES.CallMaterials.Commands.SetUrgencyCommand', {
    extend: 'SIE.cmd.Edit',
    meta: { text: "设为紧急", group: "edit"},
    canExecute: function (view) {        
        var curEntity = this.view.getCurrent();
        if (curEntity == null) { return false; }
        if (view.getSelection().length != 1) { return false; }
        var curData = curEntity.getData();
        if (curData != null && curData.Priority == 0 && (curData.Status == 0 || curData.Status == 1 || curData.Status == 3)) { return true; }
        return false;
    },
    execute: function (view, source) {
        var me = this;
        var indata = {};
        var editEntity = this.getEditEntity();
        indata.Data = Ext.encode(editEntity.data);
        var msg = Ext.String.format('确定要设为紧急吗？'.t());
        SIE.Msg.askQuestion(msg, function () {
            view.execute({
                data: indata,
                success: function (res) {
                    var errMsg = res.Result;
                    if (errMsg == '操作成功') {
                        if (view.getParent() != null)
                            view.getParent().reloadData();
                        else
                            view.reloadData();
                    }
                    else
                        SIE.Msg.showMessage(errMsg);
                }
            });
        });        
    }
});